modules/servicetemplate/src/main/java/org/eclipse/xpanse/modules/servicetemplate/ServiceTemplateManage.java

```diff
@@ -345,6 +384,21 @@ public ServiceTemplateEntity getServiceTemplateDetails(UUID id, boolean checkNam
         return existingTemplate;
     }
 
+    private void updateServiceTemplateState(boolean isAutoApproveEnabled,
```

Comment:
This method is a bit confusing from the caller perspective because sometimes we send old template entity and sometimes new template entity. Would be better if we can split this method or any other solution can be used.